ROADMAP.md

```diff
@@ -0,0 +1,579 @@
+networks, sneaker nets, isolated networks in libp2p.
+
+An undertaking like this touches upon many surfaces of the system, e.g.
+implementing new transports (Bluetooth, software defined radio,
+ultrasonic data \[Chirp\], Wi-Fi direct, QR, Ethernet, 802.11s, etc.),
+new discovery mechanisms (local peer exchange, social peer
+advertisement, etc.), packet-switched routing, and more.
+
+But it's not just about connectivity. In this new reality, nodes can
+come and go easily. As a result, new requirements will emerge, such as
+supporting *roaming* (the ability for a user to switch networks without
+dropping ongoing sessions) and *resilient state and connectivity*, to
+buffer data (potentially via a store-and-forward architecture) so that a
+node disconnecting intermittently can get up to speed when they come
+back online.
+
+For certain applications, we might want to take compensatory actions
+after a grace period, such as rebalancing data across nodes. And this
+requires us to think about group membership, partitioning, consensus,
+etc. This is just an initial train of thought, and there's a lot more
+ground to cover.
+
+**Why?** The IPFS roadmap tackles mesh networks, isolated networks,
+sneaker nets, and making IPFS work in Mars where general
+connectedness to the Internet does not exist. *The network should pave
+the way for IPFS to evolve in that direction.*
+
+### 🤫 A spyproof libp2p
+
+**What?** Supporting transports like cjdns, I2P and Tor (see
+[OpenBazaar/go-onion-transport](https://github.com/OpenBazaar/go-onion-transport))
+helps create resilient, privacy-centric communications, and moves the needle
+forward towards censorship-resistant networks. As an intermediary step, we
+should improve our existing transport stack, e.g. preventing downgrade attacks
+when securing connections.
+
+### 📱 libp2p in mobile devices
+
+**What?** While js-libp2p technically works in mobile browsers, it is
+constrained by the web sandbox. First-class integration of native libp2p
+in mobile devices could be a prerequisite for the class of offline-first
+use cases we are targeting at Protocol Labs.
+
+This implies building/supporting Java/Kotlin and Swift implementations
+-- either fully-fledged or trimmed-down versions that do the minimum
+necessary to enable other visionary goals.
+
+**Why?** Bringing libp2p natively to mobile devices is an enabler for
+offline-first use cases.
+
+### 💡 libp2p in IoT
+
+**What?** Distinct from consumer mobile devices like smartphones, IoT
+devices like embedded sensors are a potential use case for libp2p. These
+devices are typically more resource constrained and may have
+intermittent connectivity. They also tend to do minimal processing
+locally, but transmit frequently to a cloud where data analysis occurs.
+
+**Why?** IoT can greatly benefit from peer-to-peer networking. Supporting IoT
+environments with libp2p greatly expands libp2p's userbase on the one side and
```

Comment:
Are we aware of certain customers asking for this?  What has held them back from developing it?

Reply:
> Are we aware of certain customers asking for this?

Even better. There is https://github.com/Actyx built on top of rust-libp2p using libp2p in the IoT / factory space today.

// @dvc94ch @rklaehn

Reply:
interesting ROADMAP. but it seems a little broad in general to be actionable and make progress. maybe picking some clear goals for the next quarter would be more productive?